ios/BeaconScanner.swift

```diff
@@ -0,0 +1,225 @@
+///
+class BeaconScanner: NSObject, CBCentralManagerDelegate {
+  
+  var delegate: BeaconScannerDelegate?
+  
+  private var centralManager: CBCentralManager!
+  private let beaconOperationsQueue: dispatch_queue_t = dispatch_queue_create("beacon_operations_queue", nil)
+  private var shouldBeScanning: Bool = false
+  private var _urlFound = Set<NSURL>()
+  
+  var urlFound: Set<NSURL> {
+    get {
+      return _urlFound
+    }
+  }
+  
+  var urls: Array<String> {
+    get {
+      var urls = Set<String>();
+      for url in _urlFound {
+        urls.insert("\(url)");
+      }
+      return Array(urls);
+    }
+  }
+  
+  override init() {
+    super.init()
+    
+    self.centralManager = CBCentralManager(delegate: self, queue: self.beaconOperationsQueue)
+    self.centralManager.delegate = self
+    
+  }
+  
+  func startScanning() {
+    dispatch_async(self.beaconOperationsQueue) {
+      self.startScanningSynchronized()
+    }
+  }
+  
+  func stopScanning() {
+    self.centralManager.stopScan()
+  }
+  
+  deinit {
+    self.stopScanning()
+  }
+  
+  func centralManagerDidUpdateState(central: CBCentralManager)  {
+    if central.state == CBCentralManagerState.PoweredOn && self.shouldBeScanning {
+      self.startScanning();
+    }
+  }
+  
+  func centralManager(central: CBCentralManager, didDiscoverPeripheral peripheral: CBPeripheral, advertisementData: [String : AnyObject], RSSI: NSNumber) {
+    guard let serviceData = advertisementData[CBAdvertisementDataServiceDataKey] as? [NSObject : AnyObject] else { return }
+    guard isAdvertisingURL(serviceData) else { return }
+    
+    if let url = URLForFrame(serviceData) {
+      self.addURL(url)
+    }
+  }
+  
+  private func startScanningSynchronized() {
+    dispatch_async(self.beaconOperationsQueue) {
+      if self.centralManager.state != CBCentralManagerState.PoweredOn {
+        NSLog("CentralManager state is %d, cannot start scan", self.centralManager.state.rawValue)
+        self.shouldBeScanning = true
+      } else {
+        NSLog("Starting to scan for Eddystones")
+        let services = [CBUUID(string: "FEAA"), CBUUID(string: "FED8")]
+        let options = [CBCentralManagerScanOptionAllowDuplicatesKey : true]
+        self.centralManager.scanForPeripheralsWithServices(services, options: options)
+      }
+    }
+  }
+  
+  private func addURL(url: NSURL) {
+    let count = self._urlFound.count
+    self._urlFound.insert(url)
+    if self._urlFound.count != count {
+      self.delegate?.urlContextChanged(self)
+    }
+  }
+  
+  //TODO: add support for the URL eviction
+  private func removeURL(url: NSURL) {
+    self._urlFound.remove(url)
+  }
+}
+
+private func isAdvertisingURL(advertisementFrameList: [NSObject : AnyObject]) -> Bool {
+  let eddystoneUUID = CBUUID(string: "FEAA")
+  let uribeaconUUID = CBUUID(string: "FED8")
+  let data = advertisementFrameList[uribeaconUUID]
+  if (data == nil) {
```

Comment:
First we check for one kind of beacon and if not present we look for the other one